res/controllers/Hercules-DJ-Console-4-Mx-scripts.js

```diff
@@ -0,0 +1,622 @@
+//   .Several controls have the "softtakeover" option enabled, which means that the control will not move in 
+//    the application if the hardware control is far from the position in the application.
+// 5) Effects
+//   .The FX knob in the controller moves the "Fast effect" knob, which is the knob present in the Deck section and
+//    configurable in the "Equalizer" preferences in Mixxx. In a future version, it could also control the "Superknob"
+//    of the active effect.
+//   .If you keep the "shift" button pressed while moving the knob, it will move slowly (more precision).
+//   .There are 12 effects, 6 with Shift disabled and 6 with Shift enabled.
+//   .The first four effects are configured to beatloop loops of size 0.5, 1, 2 and 4. They act like the corresponding buttons in Mixxx.
+//   .If you keep the shift key pressed and press effects 1 and 2, they will set a 0.125 and 0.25th beatloops.
+//   .When a loop is set that isn't one of these four cases, buttons 3 and 4 will be lit to indicate a loop is present.
+//   .If you keep the shift key pressed and press either effects 3 or 4, It will act like pressing the loop end/reloop button.
+//   .The fifth and sixth effects are for reverse play, and reverse and roll.
+//   .From seventh to tenth, they are the first four hotcues, and act like the buttons in Mixxx (if not set, set. if set, play it)
+//   .If you keep the shift key pressed and press either of the four hotcue buttons, the hotcue will get cleared.
+//   .Eleventh and twelveth effects activate the Effects rack 1 and 2 for the specific deck.
+//
+//_____________________________________________________
+var Hercules4Mx = function() {};
+
+Hercules4Mx.userSettings = {
+    // --- DJ Console 4MX tray-icon configuration ---
+    // Midi channel for controls (Tab "advanced") If "1-2", then 0, if "3-4" then 2, if "5-6" then 4...
+    'midiChannelOffset': 0,
+    // Sensitivity (Tab "Main"). Normal means 1/1. You can change the jog wheel sensitivity by either changing this, changing the hercules configuration setting, or both.
+    'sensitivity': 1 / 1,
+
+    // --- Personal preferences configuration ---
+    // Playback speed of the virtual vinyl that is being scratched. 45.00 and 33.33 are the common speeeds. (Lower number equals faster scratch)
+    'vinylSpeed': 45,
+    // alpha value for scratching filter (start with 1/8 (0.125) and tune from there)
+    'alpha': 1 / 8,
+    // beta value for scratching filter (start with alpha/32 and tune from there)
+    'beta': (1 / 8) / 32,
+    // Indicates if the Headphone/Master mix should automatically be set to master when none of the headphone cue buttons are pressed.
+    'autoHeadMix': false,
+    // KeyRepeat speed for navigation up/down, in milliseconds. 250 is a good value. Lower values make it scroll faster
+    'naviScrollSpeed': 250,
+    // VirtualDJ-like automatic switching for headphone cue (PFL) on song load
+    'autoHeadcueOnLoad': true
```

Comment:
I suspect the autoHeadMix, autoHeadcueOnLoad, and naviScrollSpeed options are the ones most likely to be adjusted by users, so I think it would make sense to put them at the top of the options.